jetty-start/src/main/java/org/eclipse/jetty/start/CommandLineBuilder.java

```diff
@@ -54,86 +55,201 @@ public static String findJavaBin()
     }
 
     /**
-     * Perform an optional quoting of the argument, being intelligent with spaces and quotes as needed. If a subString is set in quotes it won't the subString
-     * won't be escaped.
-     *
-     * @param arg the argument to quote
-     * @return the quoted and escaped argument
-     * @deprecated no replacement, quoting is done by {@link #toQuotedString()} now.
+     * @param arg string
+     * @return Quoted string
+     * @deprecated use {@link #shellQuoteIfNeeded(String)}
      */
     @Deprecated
     public static String quote(String arg)
     {
         return "'" + arg + "'";
     }
 
-    private List<String> args;
+    private final StringBuilder commandLine = new StringBuilder();
+    private final List<String> args = new ArrayList<>();
+    private final String separator;
 
     public CommandLineBuilder()
     {
-        args = new ArrayList<String>();
+        this(false);
     }
 
+    @Deprecated
     public CommandLineBuilder(String bin)
     {
         this();
         args.add(bin);
     }
 
+    public CommandLineBuilder(boolean multiline)
+    {
+        separator = multiline ? (" \\" + System.lineSeparator() + "  ") : " ";
+    }
+
     /**
-     * Add a simple argument to the command line.
-     * <p>
-     * Will quote arguments that have a space in them.
+     * This method applies single quotes suitable for a POSIX compliant shell if
+     * necessary.
+     *
+     * @param input The string to quote if needed
+     * @return The quoted string or the original string if quotes are not necessary
+     */
+    public static String shellQuoteIfNeeded(String input)
+    {
+        // Single quotes are used because double quotes are processed differently by some shells and the xarg
+        // command used by jetty.sh
```

Comment:
```suggestion
        // Single quotes are used because double quotes are evaluated differently by some shells such as bash
        // which is used by jetty.sh and the Jetty docker image
```